modules/magento/tests/playwright/MagentoErrors.spec.js

```diff
@@ -0,0 +1,167 @@
+import { expect, test } from '@playwright/test'
+import { makeProject, wrapConsole } from '@storefront-x/testing'
+
+test('magento error handling is working', async ({ page }) => {
+  await makeProject(
+    {
+      modules: [
+        '@storefront-x/base',
+        '@storefront-x/vue',
+        '@storefront-x/vue-router-simple',
+        '@storefront-x/vue-i18n',
+        '@storefront-x/vue-pinia',
+        '@storefront-x/base-commerce',
+        '@storefront-x/graphql',
+        '@storefront-x/magento',
+        '@storefront-x/cart',
+        '@storefront-x/checkout',
+        '@storefront-x/customer',
+        '@storefront-x/customer-magento',
+        [
+          'my-module',
+          {
+            components: {
+              'Error.vue': `
+                <template></template>
+                <script setup>
+                  import MagentoError from '#ioc/errors/MagentoError'
+
+                  throw new MagentoError({
+                    message: 'Error message has been sent!'
+                  })
+                </script>
+              `,
+            },
+            pages: {
+              'index.vue': `
+                <template>
+                  <Error />
+                  <p v-if="notification.notifications">{{ notification.notifications[0].message }}</p>
+                </template>
+                <script setup>
+                  import Error from '#ioc/components/Error'
+                  import useNotificationStore from '#ioc/stores/useNotificationStore'
+
+                  const notification = useNotificationStore()
+                </script>
+              `,
+            },
+            config: {
+              'MAGENTO_URL.ts': `export default 'https://be-sfx.demo.magexo.cloud'`,
+              'VUE_I18N_LOCALES.ts': `
+                export default [
+                  {
+                    name: 'en',
+                    locale: 'en-US',
+                    prefix: '/',
+                  }
+                ]
+              `,
+            },
+          },
+        ],
+      ],
+    },
+    async ({ url }) => {
+      await wrapConsole(async () => {
+        await page.goto(url, { waitUntil: 'networkidle' })
+        await expect(await page.content()).toContain('Error message has been sent!')
```

Comment:
This is not a good assert. `page.content()` returns HTML output from the server side rendering and because playwright tests run in dev mode, thrown error in setup will show the "brown error" page instead. Or it could also be inside the JSON with transfered data. You should use `page.locator` or something similar that locates content in DOM.